### DISCUSSION TITLE
How can I perform Android BLE scanning in a BroadcastReceiver (or other equivalent)

### DISCUSSION THREAD

**Discussion by vchelaru** [shinyorg/shiny#1459] [Q&A]
I apologize if I'm going about this the wrong way, but I'm interested in performing BLE scans in the background using a BroadcastReceiver. Since the BroadcastReceiver may get raised when the app is not running, I do not have access to the MauiAppBuilder which is used to call `.UseShiny()` and `.Services.AddBluetoothLE();`

Is it possible to create/obtain an instance of an IBleManager in a BroadcastReceiver's OnReceive method? Since I've been relying on the MauiAppBuilder I don't know how to create an IBleManager any other way.

**Comment by aritchie** [comment]
Application.OnCreate will be called on a cold start of your app which is before the broadcast receiver
